### PR TITLE
k3s: 1.21.4+k3s1 -> 1.22.2+k3s2

### PR DESCRIPTION
**Pull request by JesusMtnez**
###### Motivation for this change

Upgrade k3s to latest 1.22.2 release

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/k3s/default.nix [hunk 1]
```diff
@@ -43,12 +43,12 @@ with lib;
...
+  k3sCommit = "3f5774b41eb475eb10c93bb0ce58459a6f777c5f"; # k3s git commit at the above version
+  k3sRepoSha256 = "1kjf2zkm5d3s1aj4w9gzsc3ms3a0cm900fyi9899ijczw1cbrc61";
 
-  traefikChartVersion = "9.18.2"; # taken from ./scripts/download at TRAEFIK_VERSION
-  traefikChartSha256 = "sha256-9d7p0ngyMN27u4OPgz7yI14Zj9y36t9o/HMX5wyDpUI=";
+  traefikChartVersion = "10.3.6"; # taken from ./scripts/download at TRAEFIK_VERSION
+  traefikChartSha256 = "1qzshvvy1xiakvsa56h95q64si17an6pccimzjcd7sprq4lfdbry";
 
   k3sRootVersion = "0.9.1";       # taken from ./scripts/download at ROOT_VERSION
   k3sRootSha256 = "sha256-qI84KYJKY/T6pqWZW9lOTq5NzZiu//v1zrMzUCiRTGQ=";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/k3s/default.nix:54]: Those stayed the same?

Comment (JesusMtnez) [reply pkgs/applications/networking/cluster/k3s/default.nix:54]: I have checked `k3s-root` releases and there isn't a new one.

**Comment by superherointj** [comment]
K3s now has an updateScript which is still broken. I would prefer to delay this manual upgrade so I can gather data on why it's broken. This is high on my priority list and will get sorted out. K3s will be updated as soon as the script is fixed.

**Comment by superherointj** [comment]
Update script fixed:
https://github.com/NixOS/nixpkgs/pull/140790
Once merged, let's wait and see if it works.

**Pull request closed without merging**